woocommerce/wc-admin
# Try add payment method selector to onboarding store

This PR is at an early stage to validate if I'm on the right track. I've opted to add the new selector in the `onboarding` data store instead of its own due to a `payments` data store is in the works, and adding a new one with a similar name might be confusing? (I'm totally fine to refactor all of this to its own data store, would love to get some feedback on the naming/file structure). 

Also, please advise if there's anything I've missed from the expected scope of #6877.

This PR:
- Adds a `getPaymentMethodRecommendations` selector and resolver to `onboarding` data store.
- Changes payment task to use the new selector instead of `apiFetch`.
- Does minor changes to the payment task to fix lint warnings.
- Attempts to add relevant TypeScript definitions:
    - Recommended payment method array.
    - Rule processor (inspired by [RemoteInboxNotifications](https://github.com/woocommerce/woocommerce-admin/blob/main/src/RemoteInboxNotifications/README.md)).

This PR does not:
- Convert payment task list codes to TypeScript.

### Detailed test instructions:

1. Make sure you have `remote-payment-methods` feature flag enabled in `config/development.json`
1. Install and activate https://github.com/joshuatf/remote-payments-tester.
2. Make sure the store country is set to South Africa.
3. Go to the task list and click on `Choose payment methods`.
3. Click `Set up` and note the `PayFast` fields are also included.
